File: api/src/main/java/gg/mineral/api/tick/TickLoop.java
package gg.mineral.api.tick;

public interface TickLoop {
    public static interface RollingAverage {
        /**
         * Gets the average of the rolling average.
         * 
         * @return The average of the rolling average.
         */
        double getAverage();
    }

    /**
     * Gets the 1 minute TPS.
     * 
     * @return The 1 minute TPS.
     */
    RollingAverage getTps1();

    /**
     * Gets the 5 minute TPS.
     * 
     * @return The 5 minute TPS.
     */
    RollingAverage getTps5();

    /**
     * Gets the 15 minute TPS.
     * 
     * @return The 15 minute TPS.
     */
    RollingAverage getTps15();
}

File: api/src/main/java/gg/mineral/api/tick/TickLoop.kt
package gg.mineral.api.tick

interface TickLoop {
    interface RollingAverage {
        /**
         * Gets the average of the rolling average.
         *
         * @return The average of the rolling average.
         */
        val average: Double
    }

    /**
     * Gets the 1 minute TPS.
     *
     * @return The 1 minute TPS.
     */
    val tps1: RollingAverage?

    /**
     * Gets the 5 minute TPS.
     *
     * @return The 5 minute TPS.
     */
    val tps5: RollingAverage?

    /**
     * Gets the 15 minute TPS.
     *
     * @return The 15 minute TPS.
     */
    val tps15: RollingAverage?
}
